app/models/miq_request_workflow.rb

```diff
@@ -1042,6 +1042,9 @@ def allowed_storages(_options = {})
     storages = hosts.each_with_object({}) do |host, hash|
       host.writable_storages.each { |s| hash[s.id] = s }
     end.values
+
+    storages.reject!(&:maintenance_mode)
```

Comment:
Typically, we have a method on the model at the class level like `.available_storages?` that would contain this kind of logic.  cc @gmcculloug Thoughts?